test/test_mcmc.py

```diff
@@ -125,19 +125,15 @@ def model(data):
     true_probs = np.array([0.1, 0.6, 0.3])
     data = dist.Categorical(true_probs).sample(random.PRNGKey(1), (2000,))
     init_params, potential_fn, constrain_fn = initialize_model(random.PRNGKey(2), model, data)
-    init_kernel, sample_kernel = hmc(potential_fn, algo=algo)
-    hmc_state = init_kernel(init_params,
-                            trajectory_length=1.,
-                            num_warmup=warmup_steps,
-                            progbar=False,
-                            dense_mass=dense_mass)
-    hmc_states = fori_collect(num_samples, sample_kernel, hmc_state,
-                              transform=lambda x: constrain_fn(x.z),
-                              progbar=False)
-    assert_allclose(np.mean(hmc_states['p_latent'], 0), true_probs, atol=0.02)
+    # TODO: having progbar=None just to test if that value works,
+    # change it to False when jit fori_loop issue is resolved
+    samples = mcmc(warmup_steps, num_samples, init_params, constrain_fn=constrain_fn, progbar=None,
+                   print_summary=False, potential_fn=potential_fn, algo=algo, trajectory_length=1.,
+                   dense_mass=dense_mass)
```

Comment:
@neerajprad This test exercises the new change for `mcmc` with `progbar=None` (or `False`). In my system, with dense_mass=False, algo=NUTS, this gives us 15s while the fastest speed in some recent versions are 20s. So I think the change in this PR is good to go because it works well for both of this fast-to-compile test and slow-to-compile hmm example. But we need a bit trade-off when `progbar=True`.